interceptor/interceptor.go

```diff
@@ -158,6 +162,25 @@ func initializeECDHTunnel() {
 
 		// TODO: Send an encrypted ping / confirmation to the server using the shared secret
 		// just like the 1. Syn 2. Syn/Ack 3. Ack flow in a TCP handshake
+
+		message := "I am sending Ack using shared secret!"
+
+		headers := make(map[string]interface{})
```

Comment:
Here you have a headers map, so add a custom header like maybe: 
`x-ecdh-init: true`
`x-ecdh-ack: true`

Reply:
You will then check for this header in the middleware.
```
	isECDHInit := headers.Get("x-ecdh-init").String() // is true
	if isECDHInit != "<undefined>" {
		doECDHWithClient(req, res) // so you enter here
		return nil
	}
```

and then another check in function doECDHWithClient on the middleware.go:

```
    if headers.Get("x-ecdh-ack") == 'true' {
         // register this connection as open.
    }
```

The goal is to have the ack be between the Interceptor and the Middleware. The Ack request should *not* go to the NodeJS Service Provider